src/atlas/array/helpers/ArrayForEach.h

```diff
@@ -0,0 +1,222 @@
+/*
+ * (C) Crown Copyright 2023 Met Office
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#pragma once
+
+#include <array>
+#include <tuple>
+
+#include "atlas/array/ArrayView.h"
+#include "atlas/array/Range.h"
+#include "atlas/array/helpers/ArraySlicer.h"
+#include "atlas/parallel/omp/omp.h"
+#include "atlas/util/Config.h"
+
+#include "eckit/exception/Exceptions.h"
+
+namespace atlas {
+namespace array {
+namespace helpers {
+
+namespace detail {
+
+auto sequencedConf() {
+  return util::Config("execution_policy", "sequenced");
+}
+
+auto parallelUnsequencedConf() {
+    return util::Config("execution_policy", "parallel_unsequenced");
+}
+
+enum class ExecutionPolicy {
+  parallelUnsequenced,
+  sequenced
+};
```

Comment:
I've had a `std::execution` policies. the STL classes look quite a bit more fearsome than my simple enums. Perhaps it could be some follow-on work if there's a use-case?